stable/prometheus-consul-exporter/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) ( .Values.serviceMonitor.enabled ) }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+{{- if .Values.serviceMonitor.labels }}
+  labels:
+{{ toYaml .Values.serviceMonitor.labels | indent 4}}
+{{- end }}
+  name: {{ include "prometheus-consul-exporter.fullname" . }}
+{{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+{{- end }}
+spec:
+  endpoints:
+  - targetPort: {{ .Values.service.port }}
+{{- if .Values.serviceMonitor.interval }}
+    interval: {{ .Values.serviceMonitor.interval }}
+{{- end }}
+{{- if .Values.serviceMonitor.telemetryPath }}
+    path: {{ .Values.serviceMonitor.telemetryPath }}
+{{- end }}
+{{- if .Values.serviceMonitor.timeout }}
+    scrapeTimeout: {{ .Values.serviceMonitor.timeout }}
+{{- end }}
+  jobLabel: {{ include "prometheus-consul-exporter.fullname" . }}
+  namespaceSelector:
```

Comment:
ServiceMonitors select services from the same namespace by default, so you could omit this block.

Reply:
That is right: it select services from the same namespace by default. But if you are creating the `ServiceMonitor` in different namespace ( `.Values.serviceMonitor.namespace`) then you must specify where (namespace) the service to space is. It's actually my case: I have all ServiceMonitors resources inside `monitoring` namespace but prometheus-consul-exporter aside consul (`infrastructure` namespace)